src/wp-includes/class-wp-post-type.php

```diff
@@ -440,6 +440,19 @@ public function set_props( $args ) {
 		 */
 		$args = apply_filters( 'register_post_type_args', $args, $this->name );
 
+		/**
+		 * Filters the arguments for registering a specific post type.
+		 *
+		 * The dynamic portion of the filter name, `$this->name`, refers to the post type key.
+		 *
+		 * @since 6.0.0
+		 *
+		 * @param array  $args      Array of arguments for registering a post type.
+		 *                          See the register_post_type() function for accepted arguments.
+		 * @param string $post_type Post type key.
+		 */
+		$args = apply_filters( "register_{$this->name}_post_type_args", $args, $this->name );
```

Comment:
I think it would better if the `post_type`-specific filter were applied **before** `register_post_type_args` .

The same goes for the `taxonomy`-specific filter...and both new `post_type` and `taxonomy`-specific actions.

Reply:
I do feel more strongly, tho, that where ever the post type name appears in the name of the filter, it should be done as:

```
$post_type = $this->name;
apply_filters( "register_{$post_type}_post_type_args", ... );
// or apply_filters( "register_post_type_{$post_type}_args", ... );
```

Doing that makes the Code Reference cleaner (there has been a lot of that kind of cleanup for similar filters/actions recently).